src/ObjCRuntime/Runtime.cs

```diff
@@ -276,6 +277,7 @@ unsafe static void Initialize (InitializationOptions* options)
 			usertype_cache = new Dictionary <IntPtr, bool> (IntPtrEqualityComparer);
 			intptr_ctor_cache = new Dictionary<Type, ConstructorInfo> (TypeEqualityComparer);
 			intptr_bool_ctor_cache = new Dictionary<Type, ConstructorInfo> (TypeEqualityComparer);
+			protocol_cache = new Dictionary<IntPtr, Dictionary<IntPtr, bool>> (IntPtrEqualityComparer);
```

Comment:
This could be moved inside the `if (DynamicRegistrationSupported)` check a few lines below.